docs/src/content/docs/cut-short.mdx

```diff
@@ -0,0 +1,103 @@
+---
+title: Cut-Short Requests
+---
+
+Cut-short is an Astro integration that lets you stop processing a request instantly and send back a custom response, simplifying control flow in your Astro applications. By introducing the `endRequest` function, it eliminates the need for cumbersome workarounds like bubbling up response objects, throwing and catching sentinel errors, implementing custom middleware logic or replicating error response logic across all your pages.
+
+Keep the the custom response for specific conditions close to the conditions and have it shared across all your application. It's especially useful for scenarios like user authentication and access control, where you might need to redirect users to sign-in page from anywhere that requires authentication or to turn any page they don't have access to into a 404 to avoid information leak (like GitHub does).
+
+## Installing the integration
+
+import { PackageManagers } from 'starlight-package-managers';
+
+<PackageManagers type="exec" pkg="astro" args="add @inox-tools/cut-short" />
+
+## How to use
+
+From any code that is reachable from a page rendering context can use the `endRequest` function to stop the.
+
+A page-rendering context is when you are inside of:
+
+- A middleware;
+- The frontmatter of a page component (not components _in_ the page, see [streaming](#streaming));
+- An API endpoint;
+- A function called from another page-rendering context.
+
+```ts title="src/pages/userId.ts" del={3-4} ins={7-8,18-19} {10-11}
```

Comment:
I think this example is a bit complicated, 

to simplify I'd remove the invalid example and maybe even make another file like example.ts or something where you define the getUser function and import it into the api endpoint and page.